source/adapters/hip/device.hpp

```diff
@@ -35,6 +37,13 @@ struct ur_device_handle_t_ {
 
   ~ur_device_handle_t_() noexcept(false) {
```

Comment:
Are you sure you want to obtain those values at destruction time, calling it from the constructor would appear to be more appropriate?

Reply:
Indeed!